DigitalLearningSolutions.Data/DataServices/ProgressDataService.cs

```diff
@@ -227,5 +235,98 @@ public void UnlockProgress(int progressId)
                 new { progressId }
             );
         }
+
+        public Progress? GetProgressByProgressId(int progressId)
+        {
+            return connection.Query<Progress>(
+                @"SELECT ProgressId,
+                        CandidateID,
+                        CustomisationID,
+                        Completed,
+                        RemovedDate,
+                        SystemRefreshed,
+                        SupervisorAdminID,
+                        CompleteByDate,
+                        EnrollmentMethodID,
+                        EnrolledByAdminID,
+                        SubmittedTime,
+                        CustomisationVersion,
+                        DiagnosticScore
+                    FROM Progress
+                    WHERE ProgressID = @progressId",
+                new { progressId }
+            ).SingleOrDefault();
+        }
+
+        public IEnumerable<DetailedSectionProgress> GetSectionProgressDataForProgressEntry(int progressId)
+        {
+            return connection.Query<DetailedSectionProgress>(
+                @"SELECT
+                        s.SectionID,
+                        s.SectionName,
+                        (SUM(asp1.TutStat) * 100) / (COUNT(t.TutorialID) * 2) AS Completion,
+                        SUM(asp1.TutTime) AS TotalTime,
+                        s.AverageSectionMins AS AverageTime,
+                        CASE WHEN (s.PLAssessPath IS NOT NULL AND cu.IsAssessed = 1) THEN 1 ELSE 0 END AS PostLearningAssessment,

```

Comment:
Where does this line come from? I can't see it in the original SQL.

Reply:
The only use of PLAssessPath and IsAssessed for our purposes is determining whether 'Post Learning Assessment is enabled' is true so I put it in the SQL, but following your other comment I'll pull it out into C#.